### PR TITLE
701 - Simplify APIs by using nested classes for parameter types

### PR DESCRIPTION
**Pull request by paulinea**

### CODE REVIEW

File: docs/components/ListItems.md [hunk 1]
```diff
@@ -72,11 +72,11 @@ Parameter | Default&nbsp;value | Description
...
 `modifier: Modifier` | `Modifier` | Modifier to be applied to the list item
-`icon: OdsListItemIcon?` | `null` | The leading supporting visual of the list item
+`icon: OdsListItem.Icon?` | `null` | The leading supporting visual of the list item
 `secondaryText: String?` | `null` | The secondary text of the list item
 `singleLineSecondaryText: Boolean` | `true` | Whether the secondary text is single line
 `overlineText: String?` | `null` | The text displayed above the primary text
-`trailing: OdsListItemTrailing?` | `null` | The trailing content to display at the end of the list item
+`trailing: OdsListItem.Trailing?` | `null` | The trailing content to display at the end of the list item
 `divider: Boolean` | `false` | Whether or not a divider is displayed at the bottom of the list item
 `onClick: (() -> Unit)?` | `null` | Will be called when the user clicks the list item. This parameter only has an effect if trailing is `OdsListItemTrailingIcon` or `null`.
```
Comment (florentmaitre) [inline docs/components/ListItems.md:81]:
```suggestion
`onClick: (() -> Unit)?` | `null` | Will be called when the user clicks the list item. This parameter only has an effect if trailing is `OdsListItem.TrailingIcon` or `null`.
```

File: lib/src/main/java/com/orange/ods/compose/component/snackbar/OdsSnackbar.kt
Comment (florentmaitre) [inline lib/src/main/java/com/orange/ods/compose/component/snackbar/OdsSnackbar.kt:1]: Should we nest the `OdsSnackbar` class into an `OdsSnackbarHost` object?

File: lib/src/main/java/com/orange/ods/compose/component/tab/OdsTabRow.kt [hunk 2]
```diff
@@ -79,10 +88,10 @@ private fun PreviewOdsTabRow(@PreviewParameter(OdsTabRowPreviewParameterProvider
     with(parameter) {
         OdsTabRow(
             selectedTabIndex = selectedTabIndex,
-            tabIconPosition = if (hasLeadingIconTabs) OdsTabRowTabIcon.Position.Leading else OdsTabRowTabIcon.Position.Top,
+            tabIconPosition = if (hasLeadingIconTabs) OdsTabRow.Tab.Icon.Position.Leading else OdsTabRow.Tab.Icon.Position.Top,
```
Comment (florentmaitre) [inline lib/src/main/java/com/orange/ods/compose/component/tab/OdsTabRow.kt:91]: Depending on what we do for `OdsHorizontalCard.Image.Position` (see other comment), we should have a consistent hierarchy for both enums.

File: lib/src/main/java/com/orange/ods/compose/component/tab/OdsTabsCommon.kt [hunk 3]
```diff
@@ -25,83 +25,90 @@ import com.orange.ods.compose.component.utilities.BasicPreviewParameterProvider
...
+                    icon = icon?.let { { it.Content() } },
+                    text = text?.let { { Text(text.uppercase(), maxLines = 1, overflow = TextOverflow.Ellipsis, style = OdsTheme.typography.button) } },
+                    selectedContentColor = selectedContentColor,
+                    unselectedContentColor = unselectedContentColor,
+                )
+            }
+        }
+
+        /**
+         * An icon in an [OdsTab].
```
Comment (florentmaitre) [inline lib/src/main/java/com/orange/ods/compose/component/tab/OdsTabsCommon.kt:76]:
```suggestion
         * An icon in an [OdsTabRow.Tab].
```

File: lib/src/main/java/com/orange/ods/compose/component/textfield/OdsOutlinedTextField.kt [hunk 4]
```diff
@@ -39,16 +39,16 @@ internal fun OdsOutlinedTextField(
     readOnly: Boolean = false,
     label: String? = null,
     placeholder: String? = null,
-    leadingIcon: OdsTextFieldLeadingIcon? = null,
-    trailing: OdsTextFieldTrailing? = null,
+    leadingIcon: OdsTextField.LeadingIcon? = null,
```
Comment (florentmaitre) [inline lib/src/main/java/com/orange/ods/compose/component/textfield/OdsOutlinedTextField.kt:42]: We should also update class names in `TextFields.md`.

File: lib/src/main/java/com/orange/ods/compose/component/snackbar/OdsSnackbar.kt
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/snackbar/OdsSnackbar.kt:1]: I'm not sure about this one cause `Snackbar` exists alone in Google API and `OdsSnackbar` could be used in a `SnackbarHost`. What do you think?

File: lib/src/main/java/com/orange/ods/compose/component/tab/OdsTabRow.kt (hunk 2 above)
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/tab/OdsTabRow.kt:91]: For `OdsHorizontalCard`, I remove the `Image` level to write `OdsHorizontalCard.ImagePosition` cause `Image` is not used. Here we use `OdsTabRowTab.Icon`, do you think I should create an `OdsTabRowTab.IconPosition`?
Maybe I would have to see `Image` level as object for `OdsHorizontalCard`, I don't know...

File: lib/src/main/java/com/orange/ods/compose/component/list/OdsListItem.kt [hunk 5]
```diff
@@ -293,241 +275,267 @@ private fun Modifier.rootModifier(trailing: OdsListItemTrailing?, onListItemClic
...
     /**
-     * Creates an instance of [OdsListItemIcon].
-     *
-     * @param type The type of icon.
-     * @param painter Painter of the icon.
-     * @param contentDescription The content description associated to this [OdsListItemIcon].
+     * A leading icon in an [OdsListItem].
      */
-    constructor(type: OdsListItemIconType, painter: Painter, contentDescription: String) : this(type, painter as Any, contentDescription, null)
+    class Icon private constructor(
```
[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/list/OdsListItem.kt]: Sorry, I did it in this PR...

File: lib/src/main/java/com/orange/ods/compose/component/chip/OdsChip.kt [hunk 6]
```diff
@@ -66,8 +66,8 @@ fun OdsChip(
     modifier: Modifier = Modifier,
     enabled: Boolean = true,
     selected: Boolean = false,
-    leadingIcon: OdsChipLeadingIcon? = null,
-    leadingAvatar: OdsChipLeadingAvatar? = null,
+    leadingIcon: OdsChip.LeadingIcon? = null,
+    leadingAvatar: OdsChip.LeadingAvatar? = null,
```
[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/chip/OdsChip.kt:70]: Done in this [PR](https://github.com/Orange-OpenSource/ods-android/pull/706) to merge after this one.

**Review by florentmaitre — APPROVED** [review]

**Pull request merged**